vignettes/getting-osm-data.Rmd

```diff
@@ -0,0 +1,180 @@
+waterways <- osmdata_as_sf("waterway", "river", bbox)
+waterway <- waterways$osm_multilines |>
+  filter(name == river_name) |>
+  st_intersection(st_as_sfc(bbox_expanded)) |>
+  st_transform(epsg_code) |>
+  st_geometry()
+```
+
+We also query the Overpass API for `natural:water`. The results also include features such as fountains. The geometries are not contiguous and some part of the water bodies are actually represented as lines instead of polygons. We determine and keep the only features that intersect the relevant waterway:
+
+```{r}
+# water area (polygons)
+water <- osmdata_as_sf("natural", "water", bbox)
+waterbody <- bind_rows(water$osm_polygons, water$osm_multipolygons) |>
+  st_transform(epsg_code) |>
+  st_filter(waterway, .predicate = st_intersects) |>
+  st_union() |>
+  st_geometry()
+```
+
+## 3. Street network
+
+Querying the Overpass API for the `highway` key, using the expanded bounding box to include relevant streets close to the edge of the city:
+
+```{r}
+highways_value <- c("motorway", "trunk", "primary", "secondary", "tertiary")
+
+links_value <- sapply(X = highways_value,
+                      FUN = \(x) sprintf("%s_link", x),
+                      USE.NAMES = FALSE)
+
+highways <- osmdata_as_sf("highway",
+                          c(highways_value, links_value),
+                          bbox_expanded)
+```
+
+As some of elements are returend as polygons (see e.g. [this tutorial](https://geospatial-community.netlify.app/post/2022-03-31-spatial-networks/)) we cast those too into lines and combine them with the rest of the street network:
+
+```{r}
+# cast polygons (closed streets) into lines
+poly_to_lines <- highways$osm_polygons |>
+  st_cast("LINESTRING")
+
+# combine all features in one data frame
+highways_lines <- highways$osm_lines |>
+  bind_rows(poly_to_lines) |>
+  select("highway") |>  # only keep "highway" column
+  rename(type = `highway`) |>  # rename it to "type"
+  st_transform(epsg_code)
+```
+
+## 4. Rail network
+
+Querying the Overpass API for the `railway:rail` key:tag, also using the expanded bounding box to include relevant ways close to the edge of the city:
+
+```{r}
+railways <- osmdata_as_sf("railway", "rail", bbox_expanded)
+railways_lines <- railways$osm_lines |>
+  select("railway") |>  # only keep "railway" column
+  rename(type = `railway`) |>  # rename it to "type"
+  st_transform(epsg_code)
+```
+
+## Write OSM data to file
+
+```{r}
+st_write_list <- list(
+  city_boundary = bucharest$boundary,
+  waterway = bucharest$river_centerline,
+  waterbody = bucharest$river_surface,
+  highways = bucharest$streets,
+  railways = railways_lines
+)
+
+purrr::walk(
+  st_write_list,
+  ~ st_write(
+    .x,
+    dsn = sprintf("%s_%s.gpkg",
+                  names(st_write_list)[purrr::map_lgl(st_write_list,
+                                                      identical,
+                                                      .x)], city_name),
+    append = FALSE,
+    quiet = TRUE
+  )
+)
```

Comment:
Just to be consistent with the other libraries, maybe we can add `library("purr")` at the top and  leave the library name here:

```suggestion
walk(
  st_write_list,
  ~ st_write(
    .x,
    dsn = sprintf("%s_%s.gpkg",
                  names(st_write_list)[map_lgl(st_write_list,
                                                      identical,
                                                      .x)], city_name),
    append = FALSE,
    quiet = TRUE
  )
)
```